rattan-core/src/cells/external.rs

```diff
@@ -260,93 +257,151 @@ where
+            // Make it simple as only TCP is supported.
+            let mut entry = rattan_log::TCPLogEntry::new();
+            entry.general_pkt_entry.pkt_length = pkt_len;
+            entry.general_pkt_entry.header.set_pkt_action(action as u8);
+            entry.general_pkt_entry.ts = ts;
+            entry.tcp_entry.flow_id = p.get_flow_id();
+
+            // Inspect packet buffer
+            if let Ok(ether_hdr) = etherparse::Ethernet2HeaderSlice::from_slice(p.as_slice()) {
+                #[allow(clippy::single_match)]
+                match ether_hdr.ether_type() {
+                    etherparse::EtherType::IPV4 => {
+                        match etherparse::Ipv4HeaderSlice::from_slice(
+                            p.as_slice().get(ether_hdr.slice().len()..).unwrap_or(&[]),
+                        ) {
+                            Ok(ip_hdr) => {
+                                entry.tcp_entry.ip_id = ip_hdr.identification();
+                                entry.tcp_entry.ip_frag = unsafe {
+                                    // SAFETY:
+                                    // Safe as the slice length is checked to be at least
+                                    // Ipv4Header::MIN_LEN (20) in the constructor.
+                                    u16::from_be_bytes([
+                                        *ip_hdr.slice().get_unchecked(6),
+                                        *ip_hdr.slice().get_unchecked(7),
+                                    ])
+                                };
+                                entry.tcp_entry.checksum = ip_hdr.header_checksum();
+                                #[allow(clippy::single_match)]
+                                match ip_hdr.protocol() {
+                                    etherparse::IpNumber::TCP => {
+                                        if let Ok(tcp_hdr) = etherparse::TcpHeaderSlice::from_slice(
+                                            p.as_slice()
+                                                .get(
+                                                    ether_hdr.slice().len()
+                                                        + ip_hdr.slice().len()..,
+                                                )
+                                                .unwrap_or(&[]),
+                                        ) {
+                                            entry.tcp_entry.seq = tcp_hdr.sequence_number();
+                                            entry.tcp_entry.ack = tcp_hdr.acknowledgment_number();
+                                            entry.tcp_entry.flags =
+                                                *tcp_hdr.slice().get(13).unwrap();
+                                            entry.tcp_entry.dataofs = tcp_hdr.data_offset();
+                                            let entry_bytes = entry.as_bytes().to_owned();
+                                            let _ = tx.send(RattanLogOp::Entry(entry_bytes));
```

Comment:
While avoiding blocking the data path is correct, silently swallowing errors might hide issues if the logging thread panics or the channel closes unexpectedly. Consider at least a `tracing::warn!` (rate-limited if possible) or `tracing::debug!` or a counter for dropped log entries.